goroutines panic while attach to docker 1.9, go 1.4.2

I'm building my own docker binary with debug flag "-N -l".  I am able to set break point (docker ps, for example), after attaching to the running process.  However, goroutines panic.  This is reproducible every single time.

System environment:
CentOS 7
$ uname -r
3.19.3-1.el7.elrepo.x86_64

$ go version
go version go1.4.2 linux/amd64

Docker git log:
commit bba762b19254b8315aa8d5a6a63cb00253eaa42c
Merge: e1f3a5a 6cd2d89
Author: Sebastiaan van Stijn <thaJeztah@users.noreply.github.com>
Date:   Sun Aug 16 10:59:16 2015 +0200
    Merge pull request #15237 from charleswhchan/patch-3
    Reformat /containers/(id)/resize to make it consistent with other sections.


Starting Docker daemon:
$ ./bundles/1.9.0-dev/binary/docker daemon
INFO[0000] [graphdriver] using prior storage driver "overlay"
INFO[0000] Listening for HTTP on unix (/var/run/docker.sock)
INFO[0000] Option DefaultDriver: bridge
INFO[0000] Option DefaultNetwork: bridge
INFO[0000] Firewalld running: false
INFO[0000] Loading containers: start.
..............
INFO[0000] Loading containers: done.
INFO[0000] Daemon has completed initialization
INFO[0000] Docker daemon                                 commit=bba762b-dirty execdriver=native-0.2 graphdriver=overlay version=1.9.0-dev

Attaching to docker daemon && run goroutines:
$ ps -ef | grep docker
root      8572  2138  0 01:30 pts/1    00:00:00 ./bundles/1.9.0-dev/binary/docker daemon
root      8603 21784  0 01:30 pts/72   00:00:00 grep --color=auto docker
root@xxx:~/.../github.com/derekparker/delve
$ dlv attach 8572
Type 'help' for list of commands.
(dlv) goroutines
[22 goroutines]
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4d024f]

goroutine 1 [running]:                                                                                                                                                                                                                          [22/8916]
github.com/derekparker/delve/terminal.goroutines(0x7fe850c1f668, 0xc208a14560, 0xc208a86d00, 0x0, 0x0, 0x0, 0x0)
        /root/git/go/src/github.com/derekparker/delve/terminal/command.go:237 +0x2cf
github.com/derekparker/delve/terminal.(*Term).Run(0xc2090ea480, 0x0, 0x0, 0x0)
        /root/git/go/src/github.com/derekparker/delve/terminal/terminal.go:96 +0xa80
main.execute(0x217c, 0x0, 0x0, 0x0, 0xc20803e068, 0x0)
        /root/git/go/src/github.com/derekparker/delve/cmd/dlv/main.go:312 +0x4ea
main.func·010(0xc2080a8c40, 0xc20802a9b0, 0x1, 0x1)
        /root/git/go/src/github.com/derekparker/delve/cmd/dlv/main.go:246 +0x1ec
github.com/spf13/cobra.(*Command).execute(0xc2080a8c40, 0xc20802a800, 0x1, 0x1, 0x0, 0x0)
        /root/git/go/src/github.com/spf13/cobra/command.go:534 +0x82f
github.com/spf13/cobra.(*Command).Execute(0xc2080a8000, 0x0, 0x0)
        /root/git/go/src/github.com/spf13/cobra/command.go:617 +0x414
main.main()
        /root/git/go/src/github.com/derekparker/delve/cmd/dlv/main.go:271 +0xd75

goroutine 5 [syscall]:
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 7 [chan receive, locked to thread]:
github.com/derekparker/delve/proc.(*Process).handlePtraceFuncs(0xc2080aa000)
        /root/git/go/src/github.com/derekparker/delve/proc/proc.go:676 +0x5d
created by github.com/derekparker/delve/proc.New
        /root/git/go/src/github.com/derekparker/delve/proc/proc.go:70 +0x21e

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 13 [chan receive]:
github.com/derekparker/delve/terminal.func·007()
        /root/git/go/src/github.com/derekparker/delve/terminal/terminal.go:42 +0x6d
created by github.com/derekparker/delve/terminal.(*Term).Run
        /root/git/go/src/github.com/derekparker/delve/terminal/terminal.go:48 +0x263

goroutine 11 [IO wait]:
net.(*pollDesc).Wait(0xc2097ee610, 0x72, 0x0, 0x0)
        /usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2097ee610, 0x0, 0x0)
        /usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc2097ee5b0, 0xc209b60001, 0x5ff, 0x5ff, 0x0, 0x7fe850db4e80, 0xc208a87870)
        /usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20803e070, 0xc209b60001, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/net.go:121 +0xdc
encoding/json.(*Decoder).readValue(0xc208ec8000, 0x416a00, 0x0, 0x0)
        /usr/local/go/src/encoding/json/stream.go:124 +0x5e1
encoding/json.(*Decoder).Decode(0xc208ec8000, 0x89f160, 0xc208993d00, 0x0, 0x0)
        /usr/local/go/src/encoding/json/stream.go:44 +0x7b
net/rpc/jsonrpc.(*serverCodec).ReadRequestHeader(0xc208993ce0, 0xc208a144e0, 0x0, 0x0)
        /usr/local/go/src/net/rpc/jsonrpc/server.go:66 +0x9c
net/rpc.(*Server).readRequestHeader(0xc2089b5b00, 0x7fe850c1f5e8, 0xc208993ce0, 0x0, 0x0, 0xc208a144e0, 0x410900, 0x0, 0x0)
        /usr/local/go/src/net/rpc/server.go:575 +0x97
net/rpc.(*Server).readRequest(0xc2089b5b00, 0x7fe850c1f5e8, 0xc208993ce0, 0xc208a05c80, 0xc208a14bc0, 0x7e93a0, 0x0, 0x0, 0x0, 0x0, ...)
        /usr/local/go/src/net/rpc/server.go:542 +0xa2
net/rpc.(*Server).ServeCodec(0xc2089b5b00, 0x7fe850c1f5e8, 0xc208993ce0)
        /usr/local/go/src/net/rpc/server.go:461 +0x6d
github.com/derekparker/delve/service/rpc.func·002()
        /root/git/go/src/github.com/derekparker/delve/service/rpc/server.go:65 +0x1c5
created by github.com/derekparker/delve/service/rpc.(*RPCServer).Run
        /root/git/go/src/github.com/derekparker/delve/service/rpc/server.go:66 +0x18d

goroutine 12 [IO wait]:
net.(*pollDesc).Wait(0xc2097ee5a0, 0x72, 0x0, 0x0)
        /usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2097ee5a0, 0x0, 0x0)
        /usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc2097ee540, 0xc208ed8001, 0x1dff, 0x1dff, 0x0, 0x7fe850db4e80, 0xc209b354d0)
        /usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20803e078, 0xc208ed8001, 0x1dff, 0x1dff, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/net.go:121 +0xdc
encoding/json.(*Decoder).readValue(0xc208ec8180, 0xc208a155c0, 0x0, 0x0)
        /usr/local/go/src/encoding/json/stream.go:124 +0x5e1
encoding/json.(*Decoder).Decode(0xc208ec8180, 0x89f0a0, 0xc208b942c8, 0x0, 0x0)
        /usr/local/go/src/encoding/json/stream.go:44 +0x7b
net/rpc/jsonrpc.(*clientCodec).ReadResponseHeader(0xc208b94280, 0xc2090ea540, 0x0, 0x0)
        /usr/local/go/src/net/rpc/jsonrpc/client.go:75 +0xaf
net/rpc.(*Client).input(0xc208993d40)
        /usr/local/go/src/net/rpc/client.go:109 +0xb6
created by net/rpc.NewClientWithCodec
        /usr/local/go/src/net/rpc/client.go:201 +0xd0

--- a/terminal/command.go
+++ b/terminal/command.go
@@ -234,7 +234,7 @@ func goroutines(t *Term, args ...string) error {
 	fmt.Printf("[%d goroutines]\n", len(gs))
 	for _, g := range gs {
 		prefix := "  "
-		if g.ID == state.SelectedGoroutine.ID {
+		if state.SelectedGoroutine != nil && g.ID == state.SelectedGoroutine.ID {
 			prefix = "* "
 		}
 		fmt.Printf("%sGoroutine %s\n", prefix, formatGoroutine(g))

